curvefs/src/metaserver/partition_cleaner.cpp

```diff
@@ -0,0 +1,132 @@
+/*
+ *  Copyright (c) 2021 NetEase Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * @Project: curve
+ * @Date: 2021-12-15 10:54:37
+ * @Author: chenwei
+ */
+
+#include "curvefs/src/metaserver/partition_cleaner.h"
+#include <list>
+#include "curvefs/src/metaserver/copyset/meta_operator.h"
+
+namespace curvefs {
+namespace metaserver {
+
+bool PartitionCleaner::ScanPartition() {
```

Comment:
I think is not a good way to delete inode one by one through Raft, these data are useless anymore but you pay a price to clean up the data.

objects' name for a specific fs is well-formed, they have a common prefix `fsid_`

https://github.com/opencurve/curve/blob/a193b2dfccfd098ccf2e06e5648a8e86cff90cbb/curvefs/src/common/s3util.h#L32

so, you can directly list all objects by a prefix that belongs to current fs, then delete them, after that, you can safely delete the partition.

you can check whether aws-s3-sdk support list by a prefix

https://sdk.amazonaws.com/cpp/api/LATEST/class_aws_1_1_s3_1_1_model_1_1_list_objects_result.html

Reply:
list by a prefix can not distinguish inode of partition.
it will list huge amont of object, delete these object will take a very long time.